resource/in

```diff
@@ -20,27 +21,36 @@ def get_client(dest_dir, source={}, version=None, params={}):
 
     auth0.access(ManagementAPI(source['domain']))
 
-    print(f'Fetching client {source["app-name"]} details ...')
+    log.msg("Fetching client.", name=source["app-name"])
 
     client = auth0.management.get(Client(name=source['app-name']))
 
     with open(os.path.join(dest_dir, 'credentials.yaml'), 'w') as outfile:
-        # JSON is a subset of YAML
-        json.dump({
-            'AuthProxy': {
-                'Auth0': {
-                    'ClientId': client['client_id'],
-                    'ClientSecret': client['client_secret'],
-                    'Domain': source['domain'],
+        try:
+            # JSON is a subset of YAML
+            json.dump({
+                'AuthProxy': {
+                    'Auth0': {
+                        'ClientId': client['client_id'],
+                        'ClientSecret': client['client_secret'],
+                        'Domain': source['domain'],
+                    },
                 },
-            },
-        }, outfile)
+            }, outfile)
+        except Exception as ex:
```

Comment:
If the `open()` raises it wouldn't be caught by this right? I'm not 100% if the `with` context behaves in a way that inner `try` catch its exceptions as well.

I like the idea of catching these (even if admittedly they're very very unlikely, but I guess that's why they're called "exceptions")

Reply:
I think the file descriptor will open, but the exception will only be raised when you try to interact with the file. If that's not the case, I'll just change it. ;-)

Reply:
exception handling is never pretty but yeah if we want to be defensive maybe it's better to also handle the remote possibility exception of the `open` failing 🤷‍♂️